src/main/java/org/broadinstitute/hellbender/utils/MathUtils.java

```diff
@@ -1127,39 +875,68 @@ public static double normalDistribution(final double mean, final double sd, fina
         Utils.validateArg(wellFormedDouble(mean) && wellFormedDouble(sd) && wellFormedDouble(x),
                           "mean, sd, or, x : Normal parameters must be well formatted (non-INF, non-NAN)");
 
-        double a = 1.0 / (sd * Math.sqrt(2.0 * Math.PI));
-        double b = Math.exp(-1.0 * (Math.pow(x - mean, 2.0) / (2.0 * sd * sd)));
-        return a * b;
-    }
-
-    public static double dirichletMultinomial(double[] params, int[] counts) {
-        return dirichletMultinomial(params,sum(params),counts,(int) sum(counts));
+        return (INV_SQRT_2_PI / sd) *  Math.exp(-(x - mean)*(x-mean)/ (2.0 * sd * sd));
     }
 
     /**
      * Return the likelihood of observing the counts of categories having sampled a population
      * whose categorial frequencies are distributed according to a Dirichlet distribution
-     * @param dirichletParams - params of the prior dirichlet distribution
-     * @param dirichletSum - the sum of those parameters
+     * @param params - params of the prior dirichlet distribution
      * @param counts - the counts of observation in each category
-     * @param countSum - the sum of counts (number of trials)
      * @return - associated likelihood
      */
-    public static double dirichletMultinomial(final double[] dirichletParams,
-                                              final double dirichletSum,
-                                              final int[] counts,
-                                              final int countSum) {
-        if ( dirichletParams.length != counts.length ) {
-            throw new IllegalArgumentException("The number of dirichlet parameters must match the number of categories");
+    public static double dirichletMultinomial(double[] params, int[] counts) {
+        Utils.validateArg(params.length == counts.length, "The number of dirichlet parameters must match the number of categories");
+        final double dirichletSum = sum(params);
+        final int countSum = (int) sum(counts);
+        double prefactor = log10MultinomialCoefficient(countSum,counts) + log10Gamma(dirichletSum) - log10Gamma(dirichletSum+countSum);
+        return prefactor + new IndexRange(0, counts.length).sum(n -> log10Gamma(counts[n] + params[n]) - log10Gamma(params[n]));
+    }
+
+    /**
+     * The following method implements Arrays.stream(array).map(func).toArray(), which is concise but performs poorly due
+     * to the overhead of creating a stream, especially with small arrays.  Thus we wrap the wordy but fast array code
+     * in the following method which permits concise Java 8 code.
+     *
+     * This method has been benchmarked and performs as well as array-only code.
```

Comment:
add a comment that this method *does not* modify the array but rather creates a new one. There may be applications where modifying the array in place would make sense as well.